README.md

```diff
@@ -207,23 +220,273 @@ More information can be found [here](http://www.diveintopython3.net/strings.html
+
+The `synapseutils` module in `synapsePythonClient` package has the following structure:
+```
+    module: synapseutils
+        module: copy
+            function: copy
+            function: copyWiki
+            function: copyFileHandles
+        module: sync
+            function: syncToSynapse
+            function: syncFromSynapse
+        module: monitor
+            function: notifyMe
+```
+
+### Expose all functions and classes within a Python module
+
+#### Generate .Rd files
+
+To generate the R package `synapserutils`, our first attempt is to exposing all functions under `synapseutils` Python module. In `.configure` file, we execute an R script that calls `generateRdFiles` as following:
+```r
+generateRdFiles(srcRootDir,
+                pyPkg = "synapseutils",
+                module = "synapseutils")
+```
+
+Where:
+* `srcRootDir` is the path to `synapserutils` directory. The directory must exist prior to this call. `generateRdFiles` will create a folder `auto-man` and write the generated .Rd files in this folder.
+* `pyPkg` is the name of the Python package that needs to be imported, and
+* `module` is the name of the Python module to be wrapped.
+
+#### Generate R wrappers
+
+To generate R wrappers for Python functions and constructors, we need to add an `.onLoad` hook in the R package. This hook is defined in `synapserutils/R/zzz.R` file. 
+
+In the `.onLoad` function, we add the following:
+```r
+# For the R wrappers to be available in `synapserutils` package namespace,
+# `setGeneric` must be defined in the `synapserutils` package. Therefore,
+# we need to define it in `synapserutils` and pass it through `generateRWrappers`.
+callback <- function(name, def) {
+  setGeneric(name, def)
+}
+generateRWrappers(pyPkg = "synapseutils",
+                  module = "synapseutils",
+                  setGenericCallback = callback)
+```
+
+For more information about how to use `setGeneric`, please view its reference documentation by:
+```r
+?setGeneric
+```
+
+### Expose a subset of functions within a Python module
+
+For many reasons, some Python functions are meant to be used internally, and these do not need to be exposed the R package. For example, `notifyMe` is a Python decorator, and can only be used for other Python functions.
+
+Let's omit the following functions from `synapseutils` module:
+* `copyFileHandles`
+* `notifyMe`
+
+You would need to specify a function like `selectFunctions` below in a shared .R file:
+```r
+toOmit <- c("copyFileHandles", "notifyMe")
+selectFunctions <- function(x) {
+    if (any(x$name == toOmit)) {
+        return(NULL)
+    }
+    x
+}
+```
+
+Then from the R script that generates .Rd files, update `generateRdFiles` as following:
+```r
+generateRdFiles(srcRootDir,
+                pyPkg = "synapseutils",
+                module = "synapseutils",
+                modifyFunctions = selectFunctions)
```

Comment:
Reference documentation for generateRdFiles() should explain all the function parameters.  In particular what is the definition of 'modifyFunctions'?  What are its input(s) and output(s) and what is its effect?